client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -641,9 +641,11 @@ public CompletableFuture<InsertResponse> insert(String tableName,
 
             if (!responseFuture.isCompletedExceptionally()) {
                 try {
-                    InsertResponse response = new InsertResponse(client, future.get(), clientStats);
+                    InsertResponse response = new InsertResponse(client, future.get(TIMEOUT, TimeUnit.MILLISECONDS), clientStats);
```

Comment:
should TIMEOUT also need to be configurable?

Reply:
yes, but it would change with new client. So I have not done it. Will add.

Reply:
Done.